src/Database/PG/Query/Connection.hs

```diff
@@ -37,6 +39,7 @@ module Database.PG.Query.Connection
     , PGStmtErrDetail(..)
     ) where
 
+import           Control.Concurrent.Async
 import           Control.Exception
```

Comment:
I'm a bit uneasy that we're using `Control.Exception` here, rather than [`Control.Exception.Safe`](https://hackage.haskell.org/package/safe-exceptions-0.1.7.1/docs/Control-Exception-Safe.html), and (as an extension of that) I'm also somewhat uneasy about how `MonadIO`, `Async`, and `catch` are all going to interact with each other in the presence of asynchronous exceptions.

I would have to look at how this code is being called from `graphql-engine` before I made any suggestions, but this stuff is kind of operating in a space that's been historically fraught with difficulty in Haskell (i.e. async FFI execution with exception handling) so we probably want to tread carefully.

Reply:
The newest version has a generic helper `interruptible` that aims to encapsulate all the tricky stuff (and I hope it does it right... not that sure yet, still want to test more): https://github.com/hasura/pg-client-hs/blob/1b9f06b5b84802da73f98dce9f29c3b43fe4e65c/src/Database/PG/Query/Connection.hs#L388

It uses the safe exceptions module (though not in a particularly safe way), but I suppose at least it's obvious where the dodgy parts are now.